lib/portage/dbapi/bintree.py

```diff
@@ -273,21 +306,26 @@ def unpack_contents(self, pkg, dest_dir):
 
 		pkg_path = self.bintree.getname(cpv)
 		if pkg_path is not None:
-
-			extractor = BinpkgExtractorAsync(
-				background=settings.get('PORTAGE_BACKGROUND') == '1',
-				env=settings.environ(),
-				features=settings.features,
-				image_dir=dest_dir,
-				pkg=cpv, pkg_path=pkg_path,
-				logfile=settings.get('PORTAGE_LOG_FILE'),
-				scheduler=SchedulerInterface(loop))
-
-			extractor.start()
-			yield extractor.async_wait()
-			if extractor.returncode != os.EX_OK:
-				raise PortageException("Error Extracting '{}'".format(pkg_path))
-
+			if pkg_path.endswith(SUPPORTED_XPAK_EXTENSIONS):
```

Comment:
To be honest, I think it'd preferable not to rely on suffix but instead detect the format based on contents.

Reply:
Well, I can do that, but it will use a lot of IO to open every package files.

Reply:
Now we cache the BINPKG_FORMAT for each package anyway.

Reply:
We should use `cpv._metadata['BINPKG_FORMAT']` here.